test/jasmine/tests/cone_test.js

```diff
@@ -0,0 +1,93 @@
+var Plotly = require('@lib');
+var Lib = require('@src/lib');
+
+var supplyAllDefaults = require('../assets/supply_defaults');
+var createGraphDiv = require('../assets/create_graph_div');
+var destroyGraphDiv = require('../assets/destroy_graph_div');
+var failTest = require('../assets/fail_test');
+
+describe('Test cone defaults', function() {
+    var gd;
+
+    function makeGD() {
+        return {
+            data: [{
+                type: 'cone',
+                x: [1, 2],
+                y: [1, 2],
+                z: [1, 2],
+                u: [1, 2],
+                v: [1, 2],
+                w: [1, 2]
+            }],
+            layout: {}
+        };
+    }
+
+    it('should set `visible: false` for traces missing x,y,z,u,v,w arrays', function() {
+        var keysToDelete = ['x', 'y', 'z', 'u', 'v', 'w', ''];
+
+        keysToDelete.forEach(function(k) {
+            gd = makeGD();
+            delete gd.data[0][k];
+
+            supplyAllDefaults(gd);
+            expect(gd._fullData[0].visible).toBe(!k, 'missing array ' + k);
```

Comment:
Clever, but maybe a little too compact to include `''` in this set 😈

Reply:
ok, fixed in https://github.com/plotly/plotly.js/pull/2641/commits/8dca9ae9960e3811ddd9d7d913710624ed2cb3d3